packages/console/src/pages/Applications/components/GetStarted/index.tsx

```diff
@@ -20,10 +24,57 @@ type Props = {
   onClose: () => void;
 };
 
+type DocumentFileNames = {
+  files: string[];
+};
+
+type Step = {
+  title?: string;
+  subtitle?: string;
+  metadata: string;
+};
+
 const GetStarted = ({ appName, onClose }: Props) => {
   const [isLibrarySelectorFolded, setIsLibrarySelectorFolded] = useState(false);
-  const [libraryName, setLibraryName] = useState<string>(SupportedJavascriptLibraries.Angular);
+  const [libraryName, setLibraryName] = useState<string>(SupportedJavascriptLibraries.React);
+  const [steps, setSteps] = useState<Step[]>();
+  const [activeStepIndex, setActiveStepIndex] = useState<number>(-1);
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const publicPath = useMemo(
+    () => `/console/get-started/${libraryName}/${i18next.language}`,
+    [libraryName]
+  );
+  const stepReferences = useMemo(
+    () => Array.from({ length: steps?.length ?? 0 }).map(() => React.createRef<HTMLDivElement>()),
+    [steps?.length]
+  );
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await fetch(`${publicPath}/index.json`);
+        const { files } = await response.json<DocumentFileNames>();
+        const parsedJsonData = await Promise.all(
+          files.map(async (fileName) => {
+            const response = await fetch(`${publicPath}/${fileName}`);
+            const markdownFile = await response.text();
+
+            return parseMarkdownWithYamlFrontmatter<Step>(markdownFile);
+          })
+        );
+        setSteps(parsedJsonData);
+      } catch {
+        setSteps(undefined);
+      }
```

Comment:
nit: could extract this into a hook? Could handle the response, error, and loading status easily？

Reply:
Yeah, that would be nice. Would do it in a new task
https://linear.app/silverhand/issue/LOG-2000/refactor-steps-and-markdown-loading-logics